packages/components/src/avatar/avatar.tsx

```diff
@@ -1,97 +1,356 @@
+import { Text } from '../text'
+import { tokens } from '../tokens'
+import { generateIdenticonRing } from './utils'
 
-import type { GetStyledVariants } from '@tamagui/core'
+import type { TextProps } from '../text'
+import type { ColorTokens, GetStyledVariants } from '@tamagui/core'
 
 type Variants = GetStyledVariants<typeof Base>
 
-type Props = {
-  src: string
-  size: 80 | 56 | 48 | 32 | 28 | 24 | 20 | 16
-  shape?: Variants['shape']
-  outline?: Variants['outline']
-  indicator?: GetStyledVariants<typeof Indicator>['state']
-}
+type AvatarProps =
+  | {
+      type: 'user'
+      size: 80 | 56 | 48 | 32 | 28 | 24 | 20 | 16
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      indicator?: GetStyledVariants<typeof Indicator>['state']
+    }
+  | {
+      type: 'user'
+      size: 80 | 56 | 48 | 32
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      indicator?: GetStyledVariants<typeof Indicator>['state']
+      colorHash?: number[][]
+    }
+  | {
+      type: 'group'
+      size: 80 | 48 | 32 | 28 | 20
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+    }
+  | {
+      type: 'wallet'
+      size: 80 | 48 | 32 | 28 | 20
+      name: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+    }
+  | {
+      type: 'channel'
+      size: 32 | 24 | 20
+      emoji: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      background?: ColorTokens
+      lock?: 'locked' | 'unlocked'
+    }
+  | {
+      type: 'channel'
+      size: 80
+      emoji: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      background?: ColorTokens
+    }
+  | {
+      type: 'community'
+      size: 80 | 32 | 24 | 20
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+    }
+  | {
+      type: 'account'
+      size: 80 | 48 | 32 | 28 | 24 | 20
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+    }
+  | {
+      type: 'icon'
+      size: 48 | 32 | 20
+      icon: React.ReactElement
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      color?: ColorTokens
+    }
 
 type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error'
 
-const Avatar = (props: Props) => {
-  const {
-    src,
-    size,
-    shape = 'circle',
-    outline = false,
-    indicator = 'none',
-  } = props
+// todo?: use tokens
+const paddingSizes: Record<
+  NonNullable<AvatarProps['size']>,
+  number | undefined
+> = {
+  '80': 4,
+  '56': 2,
+  '48': 2,
+  '32': 2,
+  '28': undefined,
+  '24': undefined,
+  '20': undefined,
+  '16': undefined,
+}
+
+const radiusSizes: Record<
+  NonNullable<AvatarProps['size']>,
+  number | undefined
```

Comment:
There's quite new type RadiusTokens from './tokens' in `main` if you rebase it. Instead of just "number" and same goes for values below.
16 => '$16'